### ISSUE TITLE
Automate checking against wheels on PyPI

### ISSUE THREAD

**Issue by astrofrog** [scipy/oldest-supported-numpy#16]
The requirements in setup.cfg have become complex enough that they are somewhat hard to parse and ensure that we aren't missing any cases. I wonder if we could parse these requirements and make sure that for each platform/python version/etc the version specified by setup.cfg matches the oldest available wheel on PyPI.

**Comment by rgommers** [comment]
> I wonder if we could parse these requirements and make sure that for each platform/python version/etc the version specified by setup.cfg matches the oldest available wheel on PyPI.

You're thinking in CI grab all the NumPy wheel file names, parsing them (stealing some `pip` logic or using something like https://github.com/jwodder/wheel-filename), overriding that with a list of known exceptions, and then matching everything up?

That may be a good idea. Another advantage being that we can better document why we have exceptions, most of that content for PyPy, ARM64, etc. is now only kept in PR comments.

**Comment by astrofrog** [comment]
Yes something like that! Actually now I am wondering whether we could even just generate the setup.cfg programmatically 🤔

**Comment by rgommers** [comment]
That seems a little dangerous. There'll be exceptions anyway which need to be documented, so the current way seems better than patching some programmatic output. Also, we have entries for platforms that aren't even supported by PyPI (AIX).

**Comment by rgommers** [comment]
We now have quite good test coverage, so I'll close this.
